src/librustc_typeck/collect.rs

```diff
@@ -2522,6 +2522,51 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
                     }
                 }
             }
+        } else if tcx.sess.check_name(attr, sym::instruction_set) {
+            // CHECK tcx.sess.target.target supports switching
+            if tcx.sess.target.target.options.has_thumb_interworking {
+                codegen_fn_attrs.instruction_set = match attr.meta().map(|i| i.kind) {
+                    Some(MetaItemKind::List(ref items)) => {
+                        if items.len() != 2 {
```

Comment:
Can you rewrite this check using slice patterns? I think that would look nicer

Reply:
Done :+1: